### ISSUE TITLE
Feature request: Support for Pillaged versions of tile improvement sprites

### ISSUE THREAD

**Issue by letstalkaboutdune** [yairm210/Unciv#7476]
Currently the game only informs the user of Pillaged tiles in the tile info panel on the lower-right with text such as "Mine - 2," indicating that the improvement may be repaired/completed in 2 turns. However there is no visual change to the tile sprites other than the improvement simply disappearing. I request that support for new sprites for Pillaged tiles be enabled for a more graphical indication of this status to the user.

For example, this is a set of normal Farms:
<img width="417" alt="Farm" src="https://user-images.githubusercontent.com/56904240/179447602-9adb31a9-9dd2-45c6-a8f6-e21e2fa61c41.png">

This is a set of pillaged Farms:
<img width="405" alt="Pillaged" src="https://user-images.githubusercontent.com/56904240/179447621-428a5b82-6c6c-4698-9d69-66a83dc229e4.png">

The Pillaged variant should be able to be inserted when appropriate using similar variant tags in the .png file and  .json as are already present for style and era.  

Here is a Farm-Pillaged.png to use while developing/testing:
![Farm-Pillaged](https://user-images.githubusercontent.com/56904240/179447809-4e39c7a9-f298-4c78-9f3e-85053596aee9.png)

Thank you!

**Comment by yairm210** [comment]
As usual with multi-layered images, the hard part is deciding on an order of precedence.

We already have 2 modifiers: Era and Nation, with Nation+Era only being viable if Era already exists.


Now with a new modifier, we need to face the hard questions: how do these combine?

I think that the "pillaged" modifier is a stronger one than Nation or Era, meaning it comes first.
What this looks like in practice:

- If there's a Nation-specific improvement, but a pillaged version only non-nation-specific, we use the non-specific one
- If there's a nation+era specific improvement, but only a era-specific improvement, we take that
- If there's a nation+era specific improvement, and a nation-specific improvement AND an era-specific improvement, but not one for both together, we take the era-specific - just like current improvements

In short, no matter what way we do this, there may be improvements where the nation/era variants are different from the pillaged image, because it simply doesn't exist, and we have to acknowledge that fact.

**Comment by yairm210** [comment]
Mentioning #2785 since a solution for this will solve that as well

**Comment by letstalkaboutdune** [comment]
This might be simpler than that, as Farm-Pillaged or Farm are mutually exclusive. We simply use one or the other depending on status. There will just need to be Pillaged versions for any variants that exist.

...right?

**Comment by yairm210** [comment]
I don't think you understood me.

I'm talking about if you have a special French farm, "farm-french.png", but you don't have "farm-pillaged-french.png" but only "farm-pillaged.png".
Until now it was very simple, display "farm-french.png".

But now, when it's pillaged we have 2 choices:
1. Display "farm-french.png"
2. Display "farm-pillaged.png"

I'm saying we should choose the latter. Despite the fact that the farm can change (no 'image continuity' as it were), it's still more important to show that the improvement is pillaged rather than have it be on a specific style.

**Comment by letstalkaboutdune** [comment]
I see, yes I didn't understand you because I didn't realize that nation-specific Tile Improvements sprites were possible (except for unique improvements like Polder or Terrace Farm, of course).

But I agree that when a nation-specific sprite for a non-unique improvement is pillaged, displaying a non-nation specific pillaged version is the best course of action.

**Comment by yairm210** [comment]
I do hope to get to this eventually
Currently all my time is going to bug hunts :/

**Comment by yairm210** [comment]
Okay so I need to wrap my head around the number of options we have here which is frankly ridiculous

Say we have terrain+feature+improvement+resource
So the current ranking is:
- Custom rule for (terrain+feature+improvement+resource) in tileSetConfig
- (terrain+feature+improvement+resource)
- (terrain+feature)+(improvement+resource), both of which have their own subfunctions.

improvement+resource is currently:
 - (improvement+resource)
 - (improvement)+(resource)

Which means that in total for improvements, we have 4 ranks:

- Custom rule for (terrain+feature+improvement+resource) in tileSetConfig
- (terrain+feature+improvement+resource)
- (terrain+feature)+(improvement+resource)
- (terrain+feature)+(improvement)+(resource)

So now, given we have an image for improvement-pillaged, we have 4 "new" combos - 

- Custom rule for (terrain+feature+improvement-pillaged+resource) in tileSetConfig
- (terrain+feature+improvement-pillaged+resource)
- (terrain+feature)+(improvement-pillaged+resource)
- (terrain+feature)+(improvement-pillaged)+(resource)

So we now need to sort them, as it were.

### Option 1: if there's a pillaged variant always use it

 Or in other words, order is 

- Custom rule for (terrain+feature+improvement-pillaged+resource) in tileSetConfig
- (terrain+feature+improvement-pillaged+resource)
- (terrain+feature)+(improvement-pillaged+resource)
- (terrain+feature)+(improvement-pillaged)+(resource)
- Custom rule for (terrain+feature+improvement+resource) in tileSetConfig
- (terrain+feature+improvement+resource)
- (terrain+feature)+(improvement+resource)
- (terrain+feature)+(improvement)+(resource)

## Option 2: each variant is only "better" than ones on its level


- Custom rule for (terrain+feature+improvement-pillaged+resource) in tileSetConfig
- Custom rule for (terrain+feature+improvement+resource) in tileSetConfig
- (terrain+feature+improvement-pillaged+resource)
- (terrain+feature+improvement+resource)
- (terrain+feature)+(improvement-pillaged+resource)
- (terrain+feature)+(improvement+resource)
- (terrain+feature)+(improvement-pillaged)+(resource)
- (terrain+feature)+(improvement)+(resource)

I think I'll be going with option 2 for now because otherwise pillaged versions could look radically different than non-pillaged

It'll still be a non-negligable amount of work, and this "ranking" still doesn't figure in era- or civ-specific versions.

The number of combinations rises exponentially with the number of modifiers so good luck to us all... :/

**Comment by yairm210** [comment]
AND we then need to not color in the pillaged improvement tiles... what a mess

**Comment by yairm210** [comment]
You know what? Just no.
I'm going to make it so that if this improvement has a pillaged image, then *all* the images need to be pillaged. And that's it.
There are too many options and instead of doubling the checks, we can just increase by one.

**Comment by yairm210** [comment]
![image](https://user-images.githubusercontent.com/8366208/218574952-63572e8a-174f-4098-ab96-7a7c9324e056.png)
FINALLY
